embulk-output-sqlserver/src/main/java/org/embulk/output/SQLServerOutputPlugin.java

```diff
@@ -68,6 +71,40 @@
         public InsertMethod getInsertMethod();
     }
 
+    private static class UrlAndProperties {
+        private final String url;
+        private final Properties props;
+
+        public UrlAndProperties(String url, Properties props)
+        {
+            this.url = url;
+            this.props = props;
+        }
+
+        public String getUrl()
+        {
+            return this.url;
+        }
+
+        public Properties getProps()
+        {
+            return this.props;
+        }
+
+        public Properties getPropsWithMaskedSecret()
```

Comment:
I don't think masking a password is a role of the UrlAndProperties class.
I think SQLServerOutputPlugin, who outputs logs, should mask a password.

In future we may move source codes for logging url and properties to the super class (AbstractJdbcOutputPlugin) because all subclasses do same thing.